netcheck/netcheck.go

```diff
@@ -454,11 +458,66 @@ func (c *Client) GetReport(ctx context.Context) (*Report, error) {
 		}
 	}
 
-	// TODO: if UDP is blocked, try to measure TCP connect times
-	// to DERP nodes instead? So UDP-blocked users still get a
-	// decent DERP node, rather than being randomly assigned to
-	// the other side of the planet? Or try ICMP? (likely also
-	// blocked?)
+	// Try HTTPS if UDP latency check failed for every stun server
+	// TODO: It does not show that UDP latency check failed
+	client := http.DefaultClient
+	client.Timeout = time.Second * 5
+	var rmu sync.Mutex
+	checkLatencyHTTPS := func(server string) error {
```

Comment:
This overall func is getting quite long. Can this be not a closure? (is it a closure? not obvious what it's closing over... at least a context? But an explicit context might be a win, actually)